### DISCUSSION TITLE
How many DB calls for shared data in different components?

### DISCUSSION THREAD

**Discussion by globalbuild** [remix-run/remix#3732] [General]
If components share data via a loader API call, but you do not choose to use useMatches, do 2 database calls occur? or does Remix recognize this and only make 1?

**Comment by kiliman** [comment]
Remix calls your loaders in parallel. So if you need the same data in multiple routes, you'll need to fetch those in each. Remix will not de-dupe for you.

`useMatches` is only available from your route component *after* the loaders have been called. You are able to access loader data from any route in the active route tree: root > parent > child.
